hadoop-ozone/ozone-manager/src/main/java/org/apache/hadoop/ozone/om/TrashOzoneFileSystem.java

```diff
@@ -0,0 +1,418 @@
+    return null;
+  }
+
+  @Override
+  public FSDataOutputStream create(Path path,
+      FsPermission fsPermission,
+      boolean b, int i, short i1,
+      long l, Progressable progressable) throws IOException {
+    return null;
+  }
+
+  @Override
+  public FSDataOutputStream append(Path path, int i, Progressable progressable)
+      throws IOException {
+    return null;
+  }
+
+  @Override
+  public boolean rename(Path src, Path dst) throws IOException {
+    LOG.info("Src:" + src + "Dst:" + dst);
+    RenameIterator iterator = new RenameIterator(src, dst);
+    iterator.iterate();
+    return true;
+  }
+
+  @Override
+  public boolean delete(Path path, boolean b) throws IOException {
+    DeleteIterator iterator = new DeleteIterator(path, true);
+    iterator.iterate();
+    return true;
+  }
+
+  @Override
+  public FileStatus[] listStatus(Path path) throws  IOException {
+    List<FileStatus> fileStatuses = new ArrayList<>();
+    OFSPath ofsPath = new OFSPath(path);
+    String volume = ofsPath.getVolumeName();
+    String bucket = ofsPath.getBucketName();
+    String key =  ofsPath.getKeyName();
+    String ofsPathPrefix = "/" + volume + "/" + bucket;
+    OmKeyArgs keyArgs = new OmKeyArgs.Builder()
+        .setVolumeName(volume)
+        .setBucketName(bucket)
+        .setKeyName(key)
+        .build();
+    List<OzoneFileStatus> list = ozoneManager.
+        listStatus(keyArgs, false, null, Integer.MAX_VALUE);
+    for (OzoneFileStatus status : list) {
+      Path temp = new Path(ofsPathPrefix +
+          OZONE_URI_DELIMITER + status.getKeyInfo().getKeyName());
+      FileStatus fileStatus = new FileStatus(
+          status.getKeyInfo().getDataSize(),
+          status.isDirectory(),
+          status.getKeyInfo().getFactor().getNumber(),
+          status.getBlockSize(),
+          status.getKeyInfo().getModificationTime(),
+          temp
+      );
+      fileStatuses.add(fileStatus);
+
+    }
+    return fileStatuses.toArray(new FileStatus[0]);
+  }
+
+  @Override
+  public void setWorkingDirectory(Path path) {
+
+  }
+
+  @Override
+  public Path getWorkingDirectory() {
+    return null;
+  }
+
+  @Override
+  public boolean mkdirs(Path path,
+      FsPermission fsPermission) throws IOException {
+    return false;
+  }
+
+
+  @Override
+  public FileStatus getFileStatus(Path path) throws IOException {
+    OFSPath ofsPath = new OFSPath(path);
+    String volume = ofsPath.getVolumeName();
+    String bucket = ofsPath.getBucketName();
+    String key =  ofsPath.getKeyName();
+    OmKeyArgs keyArgs = new OmKeyArgs.Builder()
+        .setVolumeName(volume)
+        .setBucketName(bucket)
+        .setKeyName(key)
+        .build();
+    String ofsPathPrefix = OZONE_URI_DELIMITER +
+        volume + OZONE_URI_DELIMITER + bucket;
+    OzoneFileStatus ofs = ozoneManager.getKeyManager().getFileStatus(keyArgs);
+    Path temp = new Path(ofsPathPrefix +
+        OZONE_URI_DELIMITER + ofs.getKeyInfo().getKeyName());
```

Comment:
this code is similar between listStatus and getFileStatus, lets have a common function to avoid duplication of code